lib/helpers/truncateByBytes.js

```diff
@@ -0,0 +1,21 @@
+export const byteCount = (string) => encodeURI(string).split(/%..|./).length - 1;
```

Comment:
Nit: It might be wise to memoize this per string since we only call it with single chars.